src/toil/test/__init__.py

```diff
@@ -253,3 +253,139 @@ def needs_cwl(test_item):
         raise
     else:
         return test_item
+
+MethodNameRegex = re.compile('[a-zA-Z_0-9]+$')
```

Comment:
Why do you anchor only at the end? Because you use match() instead of find()? That would be brittle.